airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -240,9 +241,7 @@ def _read_incremental(
             )
             record_counter = 0
             for message_counter, record_data_or_message in enumerate(records, start=1):
-                message = stream_data_to_airbyte_message(
-                    stream_name, record_data_or_message, stream_instance.transformer, stream_instance.get_json_schema()
-                )
+                message = self._get_message(record_data_or_message, stream_instance)
```

Comment:
extract logic to convert to an AirbyteMessage to `_get_message` to avoid duplication between `_read_incremental` and `_read_full_refresh`